editor.py

```diff
@@ -95,6 +95,7 @@ def edit(filename=None, contents=None, use_tty=None, suffix=''):
     if filename is None:
         tmp = tempfile.NamedTemporaryFile(suffix=suffix)
         filename = tmp.name
+        tmp.close()
```

Comment:
Do you think there any security concerns with releasing this preemptively?   Wondering if this should be restructured to write through tmp instead of just using it for the filename or, open it without a lock or exclusivity or something like that.  

Otherwise this seems fine to me.

Is this broken for all windows 10 applications or just certain editors?

Reply:
I don't think there are any security concerns. It could be restructured to write through tmp - it may be slightly more performant, but probably unnecessary. It seems you can open a file without locking it on Windows but it requires [using the win32 API](https://stackoverflow.com/a/18169890) rather than pure python.
I tested with nano and notepad - both fail. editor.py itself also crashes if the contents option is supplied.